hack/util.sh

```diff
@@ -108,6 +108,4 @@ capz::util::generate_ssh_key() {
 capz::util::ensure_azure_envs() {
     : "${AZURE_SUBSCRIPTION_ID:?Environment variable empty or not defined.}"
     : "${AZURE_TENANT_ID:?Environment variable empty or not defined.}"
-    : "${AZURE_CLIENT_ID:?Environment variable empty or not defined.}"
-    : "${AZURE_CLIENT_SECRET:?Environment variable empty or not defined.}"
```

Comment:
both `hack/ensure-azcli.sh` and `hack/ensure-acr-login.sh` currently require this field: 

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/c41c0b91e2f842adf89278b3c0fd5df765ed9827/hack/ensure-azcli.sh#L28

and 

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/c41c0b91e2f842adf89278b3c0fd5df765ed9827/hack/ensure-acr-login.sh#L36

Reply:
here is a possible change set that could address this: https://github.com/kubernetes-sigs/cluster-api-provider-azure/compare/main...jsturtevant:cluster-api-provider-azure:remove-az-cli-cred-checks?expand=1

Let me know if you want to integrate the change into this PR or have me open a separate one

Reply:
The acr change lets local users build the image with out having to manually log in.  It fixes an issue we saw when running https://github.com/kubernetes-sigs/windows-testing/pull/430 locally with a test ACR instance

Reply:
Thanks for this! To remove variables in case there are still edge cases from this change, let's do this as a follow-up PR.